### PR TITLE
FreeBSD support: GNU Make instead of BSD Make when compiling on FreeBSD

### PR DESCRIPTION
**Pull request by ajensenwaud**
Solution can be extended to pick the right make on other OS'es

### CODE REVIEW

**Comment by davkean** [comment]
@dotnet-bot retest prtest/mac/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_mac_dbg_unit32/2809/
// Retest reason: Mono deadlock

**Comment by davkean** [comment]
tag @jaredpar @TyOverby @agocke

**Comment by davkean** [comment]
tag @agocke @jaredpar @TyOverby please take a look today, this PR is 9 days old.

File: cibuild.sh [hunk 1]
```diff
@@ -55,10 +55,16 @@ done
 run_make()
 {
     local is_good=false
-    
+ 
+    MAKE="make"
+    case "$OSTYPE" in
+	    linux*)	MAKE='make';;
```
Comment (agocke) [inline cibuild.sh]: Why not combine Linux and * here?

**Comment by agocke** [comment]
Aside from comment, LGTM

**Comment by davkean** [comment]
@ajensenwaud If you update the comment, we can merge this in.

**Comment by ajensenwaud** [comment]
What part of the comment would you like me to fix?

**Comment by krytarowski** [comment]
Why do you need GNU` make`(1)? If it's needed, please add NetBSD check too.

**Comment by jasonwilliams200OK** [comment]
@krytarowski, you are right. Just tested on NetBSD7 and gmake is needed there as well. :+1: 

@mellinoe, after https://github.com/dotnet/corefx/pull/5905 is merged and corefx is independent of Mono, would it be a huge effort to make roslyn self-dependent before dotnet-core is RTMd? In case roslyn will remain relying on mono. we would need to configure mono in pkgsrc build system to roslyn, otherwise can wait a bit longer. :)

File: cibuild.sh (hunk 1 above)
Comment (jasonwilliams200OK) [inline cibuild.sh]: Or simply:

```bash
if [[ $OSTYPE == freebsd* || $OSTYPE == netbsd* ]]; then
    MAKE='gmake'
else
    Make='make'
fi
```

**Comment by jasonwilliams200OK** [comment]
@ajensenwaud, there is a minor comment: https://github.com/dotnet/roslyn/pull/8020#discussion_r51150482. if you could please fix it up then it will unblock both BSD based operating systems. Or if you prefer, I can cherry-pick your commit, amend and re-submit.

**Comment by jasonwilliams200OK** [comment]
This is addressed by #8701 and can be closed. Thanks @ajensenwaud!

**Comment by agocke** [comment]
Sounds good

**Comment by ajensenwaud** [comment]
Thanks everyone!

**Pull request closed without merging**